tests/show_techsupport/test_techsupport.py

```diff
@@ -457,3 +483,64 @@ def test_techsupport_commands(
         )
 
     pytest_assert(len(cmd_not_found) == 0, cmd_not_found)
+
+def test_secret_removed_from_show_techsupport(
+    duthosts, enum_rand_one_per_hwsku_hostname, creds_all_duts, check_image_version, setup_password
+):
+    """
+    This test checks following secrets been removed from show techsupport result:
+        Tacacs key
+        Radius key
+        snmp community string
+        /etc/shadow, which includes the hash of local/domain users' password
+        /etc/sonic/*.certs, which are Azure internal used certificates
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+
+    tacacs_passkey = creds_all_duts[duthost]['tacacs_passkey']
+    radius_passkey = creds_all_duts[duthost]['radius_passkey']
+    snmp_rocommunity = creds_all_duts[duthost]['snmp_rocommunity']
+
+    # generate a new dump file
+    duthost.shell('sudo rm -rf /var/dump/sonic_dump_*')
+    duthost.shell('sudo show techsupport')
+    dump_file_path = duthost.shell('sudo ls /var/dump/sonic_dump_* | tail -1')['stdout']
+    dump_file_name = dump_file_path.replace("/var/dump/", "")
+
+    # extract for next step check
+    duthost.shell("sudo tar -xf {0}".format(dump_file_path))
+    dump_extract_path="./{0}".format(dump_file_name.replace(".tar.gz", ""))
+    
+    # check Tacacs key
+    sed_command = "sed -nE '/secret={0}/P' {1}/etc/tacplus_nss.conf".format(tacacs_passkey, dump_extract_path)
+    check_no_result(duthost, sed_command)
+
+    sed_command = "sed -nE '/secret={0}/P' {1}/etc/pam.d/common-auth-sonic".format(radius_passkey, dump_extract_path)
+    check_no_result(duthost, sed_command)
+    
+    # check Radius key
+    sed_command = "sed -nE '/secret={0}/P' {1}/etc/radius_nss.conf".format(radius_passkey, dump_extract_path)
+    check_no_result(duthost, sed_command)
+
+    sed_command = "sed -nE '/{0}/P' {1}/etc/pam_radius_auth.conf".format(radius_passkey, dump_extract_path)
+    check_no_result(duthost, sed_command)
+    
+    # Check Radius passkey from per-server conf file /etc/pam_radius_auth.d/{ip}_{port}.conf
+    list_command = "ls {0}/etc/pam_radius_auth.d/ | grep *.conf || true".format(dump_extract_path)
```

Comment:
>grep [](http://example.com/codeflow?start=52&length=4)

Why not use wildcard in `ls` command directly?

Reply:
If in future other config files added to the pam_radius_auth.d folder, which may break this test case. so only check *.conf files

Reply:
Did you accidentally add a blank before `*`?

Reply:
Fixed